shiny/experimental/ui/_sidebar.py

```diff
@@ -0,0 +1,218 @@
+from __future__ import annotations
+
+import numbers
+import random
+from typing import Optional
+
+from htmltools import Tag, TagAttrs, TagChild, css, div, tags
+
+from shiny._typing_extensions import Literal
+
+from ._color import get_color_contrast
+from ._css import CssUnit
+from ._css import trinary as _trinary
+from ._css import validate_css_unit
+from ._fill import bind_fill_role
+from ._htmldeps import sidebar_dependency
+
+
+class Sidebar:
+    def __init__(
+        self,
+        tag: Tag,
+        collapse_tag: Optional[Tag],
+        position: Literal["left", "right"],
+        open: Literal["desktop", "open", "closed", "always"],
+        width: int,
+        max_height_mobile: Optional[str | float],
+    ):
+        self.tag = tag
+        self.collapse_tag = collapse_tag
+        self.position = position
+        self.open = open
+        self.width = width
+        self.max_height_mobile = max_height_mobile
+
+
+def sidebar(
+    *args: TagChild | TagAttrs,
+    width: int = 250,
+    position: Literal["left", "right"] = "left",
+    open: Literal["desktop", "open", "closed", "always"] = "desktop",
+    id: Optional[str] = None,
+    title: TagChild | str | numbers.Number = None,
+    bg: Optional[str] = None,
+    fg: Optional[str] = None,
+    class_: Optional[str] = None,
+    max_height_mobile: Optional[str | float] = None,
+) -> Sidebar:
+    # TODO: validate `open`, bg, fg, class_, max_height_mobile
+    # TODO: Add type annotations
+
+    if id is not None:
+        class_ = "bslib-sidebar-input" + ("" if not class_ else (" " + class_))
```

Comment:
I think this should just be:

```
extra_attrs = { "class: "bslib-sidebar-input" } 
```

and then the `extra_attrs` can be passed to the `div()` call below.

Also need to handle the `id is None` case.